src/reference/asciidoc/kafka.adoc

```diff
@@ -1610,9 +1610,42 @@ Starting with _version 2.1_, type information can be conveyed in record `Headers
 In addition, the serializer/deserializer can be configured using Kafka properties.
 
 - `JsonSerializer.ADD_TYPE_INFO_HEADERS` (default `true`); set to `false` to disable this feature on the `JsonSerializer` (sets the `addTypeInfo` property).
+- `JsonSerializer.TYPE_MAPPINGS` (default `empty`); see below
 - `JsonDeserializer.KEY_DEFAULT_TYPE`; fallback type for deserialization of keys if no header information is present.
 - `JsonDeserializer.VALUE_DEFAULT_TYPE`; fallback type for deserialization of values if no header information is present.
 - `JsonDeserializer.TRUSTED_PACKAGES` (default `java.util`, `java.lang`); comma-delimited list of package patterns allowed for deserialization; `*` means deserialize all.
+- `JsonDeserializer.TYPE_MAPPINGS` (default `empty`); see below
+
+**Mapping Types**
+
+Starting with version 2.2, you can now provide type mappings using the properties in the above list; previously you had to customize the type mapper within the serializer, deserializer.
+Mappings consist of a comma-delimited list of `token:className` pairs; on outbound, the payload's class name is mapped to the corresponding token and, on inbound, the token in the type header is mapped to the corresponding class name.
+
+For example:
+
+====
+[source, java]
+----
+senderProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+senderProps.put(JsonSerializer.TYPE_MAPPINGS, "foo:com.myfoo.Foo, bar:com.mybar.bar");
+...
+consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+consumerProps.put(JsonDeSerializer.TYPE_MAPPINGS, "foo:com.yourfoo.Foo, bar:com.yourbar.bar");
+----
+====
+
+Of course, the corresponding objects must be compatible.
+
+When using https://docs.spring.io/spring-boot/docs/2.0.4.RELEASE/reference/htmlsingle/#boot-features-kafka-extra-props[Spring Boot], these properties can be provided in the `application.properties` (or yaml) file:
+
+====
+[source]
+----
+spring.kafka.producer.value-serializer=org.springframework.kafka.support.serializer.JsonSerializer
+spring.kafka.producer.properties.spring.json.type.mapping=foo:com.myfoo.Foo,bar=com.mybar.Bar
```

Comment:
You say they have to be `colon-separate`, but here you use `=` for `bar`. I think this is typo.